lib/tirexs/http.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Tirexs.HTTP do
+  @moduledoc false
+
+
+  @doc false
+  def head(path, uri) when is_binary(path) and is_map(uri) do
+    do_request(:head, url(path, uri))
+  end
+  def head(url_or_path_or_uri) do
+    do_request(:head, url(url_or_path_or_uri))
+  end
+
+  @doc false
+  def get(path, uri) when is_binary(path) and is_map(uri) do
+    do_request(:get, url(path, uri))
+  end
+  def get(url_or_path_or_uri) do
+    do_request(:get, url(url_or_path_or_uri))
+  end
+
+  @doc false
+  def put(path, uri, body) when is_binary(path) and is_map(uri) do
+    do_request(:put, url(path, uri), body)
+  end
+  def put(path, uri) when is_binary(path) and is_map(uri) do
+    do_request(:put, url(path, uri))
+  end
+  def put(url_or_path_or_uri, body) when is_list(body) do
+    do_request(:put, url(url_or_path_or_uri), body)
+  end
+  def put(url_or_path_or_uri) do
+    do_request(:put, url(url_or_path_or_uri))
+  end
+
+  @doc false
+  def post(path, uri, body) when is_binary(path) and is_map(uri) do
+    unless body == [], do: body = to_string(body)
+    do_request(:post, url(path, uri), body)
+  end
+  def post(path, uri) when is_binary(path) and is_map(uri) do
+    do_request(:post, url(path, uri))
+  end
+  def post(url_or_path_or_uri, body) when is_list(body) do
+    unless body == [], do: body = to_string(body)
+    do_request(:post, url(url_or_path_or_uri), body)
+  end
+  def post(url_or_path_or_uri) do
+    do_request(:post, url(url_or_path_or_uri))
+  end
+
+  @doc false
+  def delete(path, uri) when is_binary(path) and is_map(uri) do
+    do_request(:delete, url(path, uri))
+  end
+  def delete(url_or_path_or_uri) do
+    do_request(:delete, url(url_or_path_or_uri))
+  end
+
+  @doc false
+  def url(path, uri) when is_binary(path) and is_map(uri) do
+    { default, given } = { Tirexs.get_uri_env(), __normalize_path__(uri) }
```

Comment:
Curiosity, this `__normalize_path__` style is for private functions?
the double `__` style